src/Npgsql/Replication/Internal/LogicalReplicationConnectionExtensions.cs

```diff
@@ -83,6 +89,8 @@ Task<ReplicationSlotOptions> CreateLogicalReplicationSlotCore()
                     slotSnapshotInitMode,
                     $"Unexpected value {slotSnapshotInitMode} for argument {nameof(slotSnapshotInitMode)}.")
             });
+            if (twoPhase)
+                builder.Append(" TWO_PHASE");
```

Comment:
Is TWO_PHASE compatible with all the snapshot options above? The compatibility syntax [in the docs](https://www.postgresql.org/docs/devel/protocol-replication.html) seems a bit weird:

> CREATE_REPLICATION_SLOT slot_name [ TEMPORARY ] { PHYSICAL [ RESERVE_WAL ] | LOGICAL output_plugin [ EXPORT_SNAPSHOT | NOEXPORT_SNAPSHOT | USE_SNAPSHOT | TWO_PHASE ] }

This makes it seem like it's not possible to do e.g. both EXPORT_SNAPSHOT and TWO_PHASE... However, the new syntax treats them as separate, compatible options:

> CREATE_REPLICATION_SLOT slot_name [ TEMPORARY ] { PHYSICAL | LOGICAL } [ ( option [, ...] ) ]

So your code makes sense to me, but it may be worth checking what's actually supported by PG15, and possibly raise a documentation inaccuracy with PG...

Reply:
> possibly raise a documentation inaccuracy with PG

Yeah, I'm even going to sent them a documentation patch.
It's already sitting here.